testing/integration/otel_test.go

```diff
@@ -1111,6 +1121,27 @@ agent:
 		2*time.Minute, 5*time.Second,
 		"Expected %d logs, got %v", numEvents, actualHits)
 
+	metricsIndex := ".ds-metrics-elastic_agent*"
+	// Check metrics from self-monitoring
+	assert.EventuallyWithT(t,
+		func(ct *assert.CollectT) {
+			findCtx, findCancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer findCancel()
+
+			docs, err := estools.GetLogsForIndexWithContext(findCtx, info.ESClient, metricsIndex, map[string]interface{}{
+				"component.id": "filestream-default",
+			})
+			require.NoError(ct, err)
+
+			actualHits.Hits = docs.Hits.Total.Value
+			output, execErr := fixture.ExecStatus(context.Background())
+			require.NoError(ct, execErr)
+			t.Logf("status output: %v", output)
+			assert.Greater(ct, actualHits.Hits, 0)
```

Comment:
This only checks that we wrote some documents, not that the document content is what it was supposed to be. I think this test would probably pass if we only wrote a JSON document that container `@timestamp` (maybe with some other minimum requirements to be a valid document).

Where are we validating that the documents this produces are what we expect and are compatible with the elastic_agent integration and the calculations the Fleet UI does for CPU and memory usage? Is that going to need a follow up?

Reply:
I see this might be coming in via https://github.com/elastic/elastic-agent/pull/7622 separately CC @mauri870

Reply:
I think @mauri870's tests cover collecting data using beats receivers, not metrics emitted by the receivers themselves.

> Where are we validating that the documents this produces are what we expect and are compatible with the elastic_agent integration and the calculations the Fleet UI does for CPU and memory usage? Is that going to need a follow up?

We're not validating it in any E2E tests added in this PR. I did do a manual test of the agent dashboard, you can see the screenshots in the PR description. I'd like to add those as a follow-up, since they're going to be big and tedious. Here, I only validated that we are, in fact, getting metrics from beats receivers at all. Is that ok?